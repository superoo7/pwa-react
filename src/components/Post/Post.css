.title {
  font-size: 3rem;
  color: #2F89DE;
}

.content {
  font-size: 1.8rem;
  line-height: 1.8;
}

.link {
  color: #2F89DE;
}

.link:focus,
.link:hover {
  color: #30699E;
  transition: color 0.3s ease;
}
