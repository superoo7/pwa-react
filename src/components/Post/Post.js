import React from 'react'
import styles from './Post.css'

import { Link } from 'react-router-dom'
const Post = (props) => (
  <div>
    <h2 className={styles.title}>
      { props.titleLink
      ? <Link className={styles.link} to={`/post/${props.slug}`}>{props.title}</Link>
      : props.title
      }
    </h2>
    <p className={styles.content}>
      {props.excerpt}
    </p>
  </div>
)

export default Post
