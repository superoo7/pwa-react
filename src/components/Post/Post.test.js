import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Post from './Post'

test('Post component should render as expected', () => {
  const component = shallow(<Post title='My post' />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Post component should render as expected', () => {
  const component = shallow(<Post title='My post' titleLink/>)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
