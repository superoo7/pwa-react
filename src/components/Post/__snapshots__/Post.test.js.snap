// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component should render as expected 1`] = `
<div>
  <h2
    className="title"
  >
    My post
  </h2>
  <p
    className="content"
  />
</div>
`;

exports[`Post component should render as expected 2`] = `
<div>
  <h2
    className="title"
  >
    <Link
      className="link"
      replace={false}
      to="/post/undefined"
    >
      My post
    </Link>
  </h2>
  <p
    className="content"
  />
</div>
`;
